dotnet/samples/GettingStartedWithProcesses/Step05/Step05_MapReduce.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Text;
+using Microsoft.SemanticKernel;
+
+namespace Step05;
+
+/// <summary>
+/// DEV HARNESS
+/// </summary>
+public class Step05_MapReduce : BaseTest
+{
+    // Target Open AI Services
+    protected override bool ForceOpenAI => true;
+
+    /// <summary>
+    /// Factor to increase the scale of the content processed.
+    /// </summary>
+    private const int ScaleFactor = 100;
+
+    private readonly string _sourceContent;
+
+    public Step05_MapReduce(ITestOutputHelper output)
+         : base(output, redirectSystemConsoleOutput: true)
+    {
+        StringBuilder content = new();
+
+        for (int count = 0; count < ScaleFactor; ++count)
+        {
+            content.AppendLine(File.ReadAllText("Grimms-The-King-of-the-Golden-Mountain.txt"));
+            content.AppendLine(File.ReadAllText("Grimms-The-Water-of-Life.txt"));
+            content.AppendLine(File.ReadAllText("Grimms-The-White-Snake.txt"));
+        }
+
+        this._sourceContent = content.ToString().ToUpperInvariant();
+    }
+
+    [Fact]
+    public async Task RunMapReduceAsync()
+    {
+        KernelProcess process = SetupMapReduceProcess(nameof(RunMapReduceAsync), "Start");
+        Kernel kernel = new();
+        using LocalKernelProcessContext localProcess =
```

Comment:
For my own education: why do we need this context here. I don't see it getting used.

Reply:
This is a feature of how the framework is designed/implemented.  One _could_ use this context to cancel or interact with the process.   We're not doing anything here...but since its disposable it needs to be handled.